lib/spack/spack/extensions.py

```diff
@@ -50,11 +50,37 @@ def load_command_extension(command, path):
     if not extension:
         return None
 
+    def ensure_package_creation(name):
+        package_name = '{0}.{1}'.format(__name__, name)
+        if package_name in sys.modules:
+            return
+
+        parts = [path] + name.split('.') + ['__init__.py']
+        init_file = os.path.join(*parts)
+        if os.path.exists(init_file):
+            m = llnl.util.lang.load_module_from_file(package_name, init_file)
+        else:
+            m = types.ModuleType(package_name)
+
+        # Setting __path__ to give spack extensions the
+        # ability to import from their own tree, see:
+        #
+        # https://docs.python.org/3/reference/import.html#package-path-rules
+        #
+        m.__path__ = [os.path.dirname(init_file)]
+        sys.modules[package_name] = m
+
+    # Create a searchable package for both the root folder of the extension
+    # and the subfolder containing the commands
+    ensure_package_creation(extension)
+    ensure_package_creation(extension + '.cmd')
```

Comment:
The two lines above create two python packages named `spack.extensions.<extension-name>` and `spack.extensions.<extension-name>.cmd`. This means that:
- The `__init__.py` file in those two extension folders is always executed when Spack imports commands
- A user can refer to python modules either in the extension root or in any subfolder of the `<ext-root>/cmd` directory